### PR TITLE
Add ESLint

### LINKED ISSUE #61: [Feature] Add ESLint
## Expected Behavior

You have to add a default linter. Preferable - ESLint

You can take help from #38

### PR DESCRIPTION
**Pull request by thepushkarp**
Fixes #61 

### Description

- Add ESLint with basic plugins like single quotes, line length, and const and let checks as suggested by @yashLadha and @aashutoshrathi 
- Removed semistandard and snazzy from dependencies as they were not used after adding eslint
- Add pretest before push

### Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### Non Functional Requirement
- [x] Follows the code style of this project.
- [x] Tests Cover Changes
- [x] I have performed a self-review of my own code
- [x] All new and existing tests passed.
- [ ] Documentation

### CODE REVIEW

**Comment by aashutoshrathi** [comment]
/remind me to merge this in 1 day. 😄

**Comment by aashutoshrathi** [comment]
/remind me to merge this in 1 day

**Comment by yashLadha** [comment]
Please resolve conflicts @thepushkarp

File: public/client.js [hunk 1]
```diff
@@ -5,44 +5,44 @@
 // add other scripts at the bottom of index.html
 
 $('#username').keyup(function () {
-  let userfield = $('#username');
-  let profile = 'https://api.github.com/users/' + userfield.val();
+  let userfield = $('#username')
```
Comment (yashLadha) [inline public/client.js]: should be `const` as it is not changing later in the program

**Comment by thepushkarp** [comment]
@aashutoshrathi @yashLadha 
This PR changs 'no-var' to 'prefer-const
Attached is the output of `npm run lint`

Can you review and suggest changes that are to be made?

Screenshot from 2019-04-24 02-26-23

**Comment by yashLadha** [comment]
Is `semi` needed @aashutoshrathi ?

**Comment by yashLadha** [comment]
Else looks good @thepushkarp  hope you learnt from this PR

**Pull request merged**